### PR TITLE
LPS-120006 Performace issues with segments due to excesive Elasticsearch queries

### PR DESCRIPTION
**Pull request by darquesdev**
**Motivation**:
The aim of this PR is to improve the current performance in the segments engine when the segments of an user need to be computed

**Approach**:
Once the segments of an user are being calculated and persisted in DB let's get them directly from there. In addition to this, these queries are going to use the database cache. 

**How to test it**:
A couple of scripts were attached in this ticket https://issues.liferay.com/browse/PTR-1878:
- 1st for [creating many segments](https://issues.liferay.com/secure/attachment/369867/createManySegments.groovy)
- 2nd for for [testing the timing](https://issues.liferay.com/secure/attachment/369866/testTimingsSegments.groovy) calculating the segments which a user belongs to

Note:
Since this piece of code is critical in segments computing, we consider very important:
- A QA validation before sending this PR (@CarlosBrichete)
- Share the solution with Support since it might be backported to 7.2 (@ricardocousolr, @jorgediaz-lr)

CC: @cgoncas

### CODE REVIEW

**Comment by darquesdev** [comment]
ci:test:relevant

File: modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java [hunk 1]
```diff
@@ -60,7 +61,9 @@
 	@ClassRule
 	@Rule
 	public static final AggregateTestRule aggregateTestRule =
-		new LiferayIntegrationTestRule();
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			SynchronousDestinationTestRule.INSTANCE);
```
Comment (cgoncas) [inline modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java:66]: I think in the last implementation, SynchronousDestinationTestRule was not needed but @Sync was

Comment (darquesdev) [reply modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/provider/test/DefaultSegmentsEntryProviderTest.java:66]: that's right, thanks

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by ricardocousolr** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by dgarciasarai** [comment]
ci:test:sf

**Pull request closed without merging**